gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -209,7 +207,9 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
           // Dataset Output path is injected in each copyableFile.
           // This can be optimized by having a dataset level equivalent class for copyable entities
           // and storing dataset related information, e.g. dataset output path, there.
-          if (!fileSetRoot.isPresent()) {
+
+          // Currently datasetOutputPath is only present for hive datasets.
+          if (!fileSetRoot.isPresent() && copyableFile.getDatasetOutputPath() != null) {
```

Comment:
Is the solution to add this check or to add a datasetOutputPath to all workunits?

Reply:
We planned to provide datasetOutputPath to only hive workunits.